File: algorithms/sorting/quick_sort.c
#include<stdio.h>
#define MAX 20

int partition(int arr[],int lb,int ub)
{
int pivot=arr[lb],up=ub,down=lb,temp=0;
while(up>down)
	{
	while(arr[down]<=pivot && down<ub)
		{down++;
		}
	while(arr[up]>pivot)
		{up--;
		}
	if(up>down)
		{temp=arr[down];
		arr[down]=arr[up];
		arr[up]=temp;
		}
	}
arr[lb]=arr[up];
arr[up]=pivot;
return up;
}	




void quicksort(int arr[],int lb,int ub)
{
if(lb>=ub)
	return;
int j;
j=partition(arr,lb,ub);
quicksort(arr,lb,j-1);
quicksort(arr,j+1,ub);

return;
}


void main()
{
int n,arr[MAX],i;
printf("enter no of elements\n");
scanf("%d",&n);
printf("\nenter the elements\n");
for(i=0;i<n;i++)
	scanf("%d",&arr[i]);
quicksort(arr,0,n-1);
printf("\n sorted array\n");
for(i=0;i<n;i++)
	printf("%d \t",arr[i]);
}

File: algorithms/sorting/quick_sort.py


def partition(array, low, high):
    pivot = array[high]
    i = low - 1
    for j in range(low,high):
        if array[j] < pivot:
            i += 1
            array[i], array[j] = array[j], array[i]
            
    array[i+1], array[high] = array[high], array[i+1]
    return i + 1, array


def quick_sort(array, low, high):
    if low < high:
        partition_index, array = partition(array, low, high)
        quick_sort(array, low, partition_index-1)
        quick_sort(array, partition_index+1, high)
    
    return array


if __name__ == "__main__":
    array = input("Enter the numbers")
    array = [int(num) for num in array.split(" ")]
    array = quick_sort(array, 0, len(array)-1)
    print(array)